subprojects/core/src/main/java/org/gradle/internal/buildtree/BuildTreeScopeServices.java

```diff
@@ -163,4 +166,9 @@ protected BuildLogicBuildQueue createBuildLogicBuildQueue(
     ) {
         return new DefaultBuildLogicBuildQueue(fileLockManager, buildStateRegistry, buildTreeWorkGraphController);
     }
+
+    private File buildExpansionCacheDir(ProjectCacheDir projectCacheDir) {
+        File rootExpansionCacheDir = new File(projectCacheDir.getDir(), DecompressionCache.EXPANSION_CACHE_KEY);
+        return new File(rootExpansionCacheDir, "settings");
```

Comment:
🤔 Like I say below, do we care if the locking is shared across the entire build?

If I'm following this correctly, this also looks like it's scoped to the entire build tree, so this would cause every build that's extracting zips/tars from a settings context to block on one another. 

I think this means we'd want to push `DecompressionCacheFactory` down to build scope and not build tree.  We can probably then remove the extra "settings" path because once we're extracting zips and tars for a project, we can no longer extract something for settings.  We don't need the extra separation.

Reply:
I think this is still needed at this scope.  The problem with using `settings` as the content dir if there was potentially also a subproject named `settings` should be addressed by the change described above.